evil-commands.el

```diff
@@ -1623,12 +1623,12 @@ of the block."
 
 (evil-define-command evil-undo (count)
   "Undo COUNT changes in buffer using `evil-undo-function'."
-  (interactive "<c>")
+  (interactive "p")
```

Comment:
Perhaps `(interactive "*p")` (error if buffer is read-only)?

Reply:
Hm, it doesn't seem to ever hit that case sinde read-only buffers tend to not have any undo data in the first place or allowing you to edit them to create it.

Reply:
Ah ok. A user could potentially enable `read-only-mode` after having made some changes though. I just noticed that `undo-redo` used `"*p"`.

Reply:
Fair enough, fixed.